audio/midi/midifile.cpp

```diff
@@ -135,8 +135,14 @@ void MidiFile::writeEvent(const MidiEvent& event)
             case ME_META:
                   put(ME_META);
                   put(event.metaType());
-                  putvl(event.len());
-                  write(event.edata(), event.len());
+                  if (event.metaType() == META_TRACK_NAME) {
```

Comment:
Why track name type only? What about the other text types (0x01 - 0x07)?

Reply:
Because the issue is about track name only I guess, "Sequence Track Name MIDI messages wrongly null-terminated"

Reply:
Obviously, we have to check out what the MIDI standard says here.

Reply:
In https://www.midi.org/component/edocman/rp-001-v1-0-standard-midi-files-specification-96-1-4-pdf/fdocument?Itemid=9999, it looks like types 0x01–0x07 are text types, and https://www.midi.org/component/edocman/smf-device-name-and-program-name-meta-events/fdocument?Itemid=9999 adds 0x08 and 0x09. In https://github.com/musescore/MuseScore/blob/6e287410b52e4173aa21109e1ca11bd563978feb/audio/midi/event.h#L77-L81,  we define some "mscore extensions" which are also (I believe) text types. Should we null terminate these extensions? Also, tangentially related, would it make sense if we changed the (nonstandard) extensions to be sent as System Exclusive messages or sequencer specific meta messages, instead?